2025-10-31 12:42:13,768 : INFO : root : Loading config.
2025-10-31 12:42:13,772 : INFO : YamlMultifileConfigLoader : Machine config file #1: config.yaml
2025-10-31 12:42:13,773 : INFO : ConfigProcessor : Loading config from cache: C:\Users\antho\AppData\Local\Temp\fdca14d0c9d658ab8200d57120800af5.mpf_cache
2025-10-31 12:42:13,922 : WARNING : ConfigProcessor : Config file in cache changed: C:\Users\antho\nightpin\modes\attract\config\attract.yaml
2025-10-31 12:42:13,922 : INFO : ConfigProcessor : Loading config from file C:\Users\antho\mpf\Lib\site-packages\mpf\modes\attract\config\attract.yaml.
2025-10-31 12:42:13,925 : INFO : ConfigProcessor : Loading config: C:\Users\antho\mpf\Lib\site-packages\mpf\modes\attract\config\attract.yaml
2025-10-31 12:42:13,926 : INFO : ConfigProcessor : Loading config from file C:\Users\antho\nightpin\modes\attract\config\attract.yaml.
2025-10-31 12:42:13,938 : INFO : ConfigProcessor : Loading config: C:\Users\antho\nightpin\modes\attract\config\attract.yaml
2025-10-31 12:42:13,939 : INFO : ConfigProcessor : Config file cache created: C:\Users\antho\AppData\Local\Temp\5dec96b1547c4adf5bdd3d1715e74bc0.mpf_cache
2025-10-31 12:42:13,939 : INFO : ConfigProcessor : Loading config from cache: C:\Users\antho\AppData\Local\Temp\202a5840610e95051f9f473dfa660cc0.mpf_cache
2025-10-31 12:42:13,940 : INFO : ConfigProcessor : Loading config from cache: C:\Users\antho\AppData\Local\Temp\5dec96b1547c4adf5bdd3d1715e74bc0.mpf_cache
2025-10-31 12:42:13,951 : INFO : ConfigProcessor : Loading config from cache: C:\Users\antho\AppData\Local\Temp\2cd13a2d3f375e00ab01d6d09a85d8e7.mpf_cache
2025-10-31 12:42:13,963 : INFO : ConfigProcessor : Loading config from cache: C:\Users\antho\AppData\Local\Temp\9220b44b97b91972524500b81035d790.mpf_cache
2025-10-31 12:42:13,964 : INFO : ConfigProcessor : Loading config from cache: C:\Users\antho\AppData\Local\Temp\31ee881584fa6fce665a672a4f3f633a.mpf_cache
2025-10-31 12:42:13,965 : INFO : ConfigProcessor : Loading config from cache: C:\Users\antho\AppData\Local\Temp\5e841763ec73085bbf92358fe6afda21.mpf_cache
2025-10-31 12:42:13,966 : INFO : ConfigProcessor : Loading config from cache: C:\Users\antho\AppData\Local\Temp\7f73d12aa9beccac1cbd8c265ccda72e.mpf_cache
2025-10-31 12:42:13,966 : INFO : ConfigProcessor : Loading config from cache: C:\Users\antho\AppData\Local\Temp\a03ebaa9d6d5676c418deb142cb34911.mpf_cache
2025-10-31 12:42:13,967 : INFO : ConfigProcessor : Loading config from cache: C:\Users\antho\AppData\Local\Temp\b17152efa55a24f0424602723f8d3921.mpf_cache
2025-10-31 12:42:13,967 : INFO : ConfigProcessor : Loading config from cache: C:\Users\antho\AppData\Local\Temp\e87fc0e7d2f8a851b5fa3f604e0d0c6e.mpf_cache
2025-10-31 12:42:13,969 : INFO : ConfigProcessor : Loading config from cache: C:\Users\antho\AppData\Local\Temp\196ac76b6cab3a9880804c826cfbcb5c.mpf_cache
2025-10-31 12:42:13,969 : INFO : ConfigProcessor : Loading config from cache: C:\Users\antho\AppData\Local\Temp\55437eb303afaa980597ff60f339a871.mpf_cache
2025-10-31 12:42:13,970 : INFO : ConfigProcessor : Loading config from cache: C:\Users\antho\AppData\Local\Temp\fddc1006228d3387e6d1e46051a3638d.mpf_cache
2025-10-31 12:42:13,970 : INFO : ConfigProcessor : Loading config from cache: C:\Users\antho\AppData\Local\Temp\834292dd7853e12a66732dc89828c828.mpf_cache
2025-10-31 12:42:13,971 : INFO : ConfigProcessor : Loading config from cache: C:\Users\antho\AppData\Local\Temp\46f8fa3e41dfeeb8232af2f03a13b6c4.mpf_cache
2025-10-31 12:42:13,972 : INFO : ConfigProcessor : Loading config from cache: C:\Users\antho\AppData\Local\Temp\c67e9c175da6c060302945929b27b878.mpf_cache
2025-10-31 12:42:13,973 : INFO : Machine : Mission Pinball Framework Core Engine v0.80.0.dev11
2025-10-31 12:42:13,973 : INFO : Machine : Command line arguments: {'no_load_cache': False, 'create_config_cache': True, 'bcp': False, 'configfile': ['config.yaml'], 'force_assets_load': False, 'platform_integration_test': False, 'jsonlogging': False, 'logfile': 'logs\\2025-10-31-12-42-13-mpf-LAPTOP-MPGP6R69.log', 'pause': False, 'production': False, 'text_ui': False, 'loglevel': 15, 'consoleloglevel': 20, 'force_platform': 'virtual', 'syslog_address': None, 'full_logfile_path': 'C:\\Users\\antho\\nightpin\\logs\\2025-10-31-12-42-13-mpf-LAPTOP-MPGP6R69.log'}
2025-10-31 12:42:13,973 : INFO : Machine : MPF path: C:\Users\antho\mpf\Lib\site-packages\mpf
2025-10-31 12:42:13,973 : INFO : Machine : Machine path: C:\Users\antho\nightpin
2025-10-31 12:42:13,973 : INFO : Machine : Platform: win32
2025-10-31 12:42:13,973 : INFO : Machine : Python executable location: C:\Users\antho\mpf\Scripts\python.exe
2025-10-31 12:42:13,973 : INFO : Machine : Python version: 3.12.7 (64-bit)
2025-10-31 12:42:13,975 : INFO : Machine : Initializing MPF...
2025-10-31 12:42:14,090 : INFO : EventManager : Event: ======'machine_var_credits_string'====== Args={'value': 'FREE PLAY', 'prev_value': None, 'change': True}
2025-10-31 12:42:14,121 : INFO : EventManager : Event: ======'machine_var_master_volume'====== Args={'value': 0.5, 'prev_value': None, 'change': True}
2025-10-31 12:42:14,121 : INFO : EventManager : Event: ======'machine_var_player1_score'====== Args={'value': 2200, 'prev_value': None, 'change': True}
2025-10-31 12:42:14,121 : INFO : EventManager : Event: ======'machine_var_player2_score'====== Args={'value': 1200, 'prev_value': None, 'change': True}
2025-10-31 12:42:14,121 : INFO : EventManager : Event: ======'machine_var_mpf_version'====== Args={'value': 'Mission Pinball Framework v0.80.0.dev11', 'prev_value': None, 'change': True}
2025-10-31 12:42:14,121 : INFO : EventManager : Event: ======'machine_var_mpf_extended_version'====== Args={'value': 'Mission Pinball Framework v0.80.0.dev11, Config version:6, Show version: 6, BCP version:1.1', 'prev_value': None, 'change': True}
2025-10-31 12:42:14,121 : INFO : EventManager : Event: ======'machine_var_python_version'====== Args={'value': '3.12.7', 'prev_value': None, 'change': True}
2025-10-31 12:42:14,308 : INFO : EventManager : Event: ======'machine_var_platform'====== Args={'value': 'Windows-11-10.0.26100-SP0', 'prev_value': None, 'change': True}
2025-10-31 12:42:14,308 : INFO : EventManager : Event: ======'machine_var_platform_system'====== Args={'value': 'Windows', 'prev_value': None, 'change': True}
2025-10-31 12:42:14,308 : INFO : EventManager : Event: ======'machine_var_platform_release'====== Args={'value': '11', 'prev_value': None, 'change': True}
2025-10-31 12:42:14,308 : INFO : EventManager : Event: ======'machine_var_platform_version'====== Args={'value': '10.0.26100', 'prev_value': None, 'change': True}
2025-10-31 12:42:14,308 : INFO : EventManager : Event: ======'machine_var_platform_machine'====== Args={'value': 'AMD64', 'prev_value': None, 'change': True}
2025-10-31 12:42:14,308 : INFO : EventManager : Event: ======'machine_var_log_file_path'====== Args={'value': 'C:\\Users\\antho\\nightpin\\logs\\2025-10-31-12-42-13-mpf-LAPTOP-MPGP6R69.log', 'prev_value': None, 'change': True}
2025-10-31 12:42:14,308 : INFO : EventManager : Event: ======'init_phase_1'====== Args={}
2025-10-31 12:42:14,351 : INFO : EventManager : Event: ======'drop_target_bank_right_bank_up'====== Args={}
2025-10-31 12:42:14,351 : INFO : EventManager : Event: ======'drop_target_bank_left_bank_up'====== Args={}
2025-10-31 12:42:14,358 : INFO : EventManager : Event: ======'init_phase_2'====== Args={}
2025-10-31 12:42:14,365 : ERROR : asyncio : Exception in callback EventManager._queue_task_done(<Task finishe...e_immediate')>)
handle: <Handle EventManager._queue_task_done(<Task finishe...e_immediate')>)>
Traceback (most recent call last):
  File "C:\Users\antho\AppData\Local\Programs\Python\Python312\Lib\asyncio\events.py", line 88, in _run
    self._context.run(self._callback, *self._args)
  File "C:\Users\antho\mpf\Lib\site-packages\mpf\core\events.py", line 795, in _queue_task_done
    future.result()
  File "C:\Users\antho\mpf\Lib\site-packages\mpf\core\events.py", line 708, in _run_handlers_sequential
    handler.callback(queue=queue, **merged_kwargs)
  File "C:\Users\antho\mpf\Lib\site-packages\mpf\core\mode_controller.py", line 97, in initialize_modes
    item.method(config=mode.config[item.config_section],
  File "C:\Users\antho\mpf\Lib\site-packages\mpf\core\config_player.py", line 150, in process_mode_config
    config = self.validate_config(config)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\antho\mpf\Lib\site-packages\mpf\core\config_player.py", line 113, in validate_config
    validated_config[event] = self.validate_config_entry(settings, event)
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\antho\mpf\Lib\site-packages\mpf\config_players\device_config_player.py", line 56, in validate_config_entry
    raise e
  File "C:\Users\antho\mpf\Lib\site-packages\mpf\config_players\device_config_player.py", line 52, in validate_config_entry
    self._validate_config_item(device, device_settings))
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\antho\mpf\Lib\site-packages\mpf\config_players\slide_player.py", line 21, in _validate_config_item
    device_settings = self.machine.config_validator.validate_config(
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\antho\mpf\Lib\site-packages\mpf\core\config_validator.py", line 164, in validate_config
    return self._validate_config(config_spec, source, base_spec=base_spec, add_missing_keys=add_missing_keys,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\antho\mpf\Lib\site-packages\mpf\core\config_validator.py", line 204, in _validate_config
    processed_config[k] = self.validate_config_item(
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\antho\mpf\Lib\site-packages\mpf\core\config_validator.py", line 243, in validate_config_item
    return self.validate_item(item, validation, validation_failure_info)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\antho\mpf\Lib\site-packages\mpf\core\config_validator.py", line 699, in validate_item
    return self.validator_list[validator](item, validation_failure_info=validation_failure_info, param=param)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\antho\mpf\Lib\site-packages\mpf\core\config_validator.py", line 390, in _validate_type_enum
    raise self.validation_error(
mpf.exceptions.config_file_error.ConfigFileError: Config File Error in ConfigValidator: Failed to load config player mode_attract_will_stop:slides with settings {'attract': {'action': 'stop', 'target': None, 'priority': None, 'expire': None, 'slide': None, 'tokens': {}}} >> Config validation error: Entry slide_player:slide_player:action = "stop" is not valid. Entry "stop" is not valid for enum. Valid values are: play,remove,method,animation,update,preload,queue,queue_first,queue_immediate Error Code: CFE-ConfigValidator-5 (https://missionpinball.org/logs)
2025-10-31 12:42:14,389 : INFO : Machine : Starting the main run loop with active modes: []
2025-10-31 12:42:14,389 : WARNING : Machine : Attract mode is not active, game will not be playable. Please check your attract mode configuration.
2025-10-31 12:42:14,390 : INFO : Machine : Shutting down...
2025-10-31 12:42:14,390 : INFO : EventManager : Event: ======'shutdown'====== Args={}
2025-10-31 12:42:15,402 : ERROR : Machine : Runtime Exception
Traceback (most recent call last):
  File "C:\Users\antho\mpf\Lib\site-packages\mpf\core\machine.py", line 810, in _run_loop
    raise self._exception['exception']
  File "C:\Users\antho\AppData\Local\Programs\Python\Python312\Lib\asyncio\events.py", line 88, in _run
    self._context.run(self._callback, *self._args)
  File "C:\Users\antho\mpf\Lib\site-packages\mpf\core\events.py", line 795, in _queue_task_done
    future.result()
  File "C:\Users\antho\mpf\Lib\site-packages\mpf\core\events.py", line 708, in _run_handlers_sequential
    handler.callback(queue=queue, **merged_kwargs)
  File "C:\Users\antho\mpf\Lib\site-packages\mpf\core\mode_controller.py", line 97, in initialize_modes
    item.method(config=mode.config[item.config_section],
  File "C:\Users\antho\mpf\Lib\site-packages\mpf\core\config_player.py", line 150, in process_mode_config
    config = self.validate_config(config)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\antho\mpf\Lib\site-packages\mpf\core\config_player.py", line 113, in validate_config
    validated_config[event] = self.validate_config_entry(settings, event)
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\antho\mpf\Lib\site-packages\mpf\config_players\device_config_player.py", line 56, in validate_config_entry
    raise e
  File "C:\Users\antho\mpf\Lib\site-packages\mpf\config_players\device_config_player.py", line 52, in validate_config_entry
    self._validate_config_item(device, device_settings))
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\antho\mpf\Lib\site-packages\mpf\config_players\slide_player.py", line 21, in _validate_config_item
    device_settings = self.machine.config_validator.validate_config(
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\antho\mpf\Lib\site-packages\mpf\core\config_validator.py", line 164, in validate_config
    return self._validate_config(config_spec, source, base_spec=base_spec, add_missing_keys=add_missing_keys,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\antho\mpf\Lib\site-packages\mpf\core\config_validator.py", line 204, in _validate_config
    processed_config[k] = self.validate_config_item(
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\antho\mpf\Lib\site-packages\mpf\core\config_validator.py", line 243, in validate_config_item
    return self.validate_item(item, validation, validation_failure_info)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\antho\mpf\Lib\site-packages\mpf\core\config_validator.py", line 699, in validate_item
    return self.validator_list[validator](item, validation_failure_info=validation_failure_info, param=param)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\antho\mpf\Lib\site-packages\mpf\core\config_validator.py", line 390, in _validate_type_enum
    raise self.validation_error(
mpf.exceptions.config_file_error.ConfigFileError: Config File Error in ConfigValidator: Failed to load config player mode_attract_will_stop:slides with settings {'attract': {'action': 'stop', 'target': None, 'priority': None, 'expire': None, 'slide': None, 'tokens': {}}} >> Config validation error: Entry slide_player:slide_player:action = "stop" is not valid. Entry "stop" is not valid for enum. Valid values are: play,remove,method,animation,update,preload,queue,queue_first,queue_immediate Error Code: CFE-ConfigValidator-5 (https://missionpinball.org/logs)
2025-10-31 12:42:15,405 : INFO : root : MPF run loop ended.
